### DISCUSSION TITLE
Having a simpler way to secure endpoints with Auth (I'm using Auth0)

### DISCUSSION THREAD

**Discussion by pbiggar** [darklang/dark#4348] [Feedback]
A user says:
>I've had a chance to play with Dark a lot for the last couple of days and a few things I want to discuss:
>- Having a simpler way to secure endpoints with Auth (I'm using Auth0). I'm currently doing this as per Ian Smith's recommendation on Slack - I'll have to add this to all my endpoints. Maybe there already is a more succinct way to do this. I'm imagining a one-liner. If not, do you have any plans as to how you want to improve this?
>- Storing secrets somewhere. I guess I could store them in a Datastore but just doesn't feel like the right abstraction. Especially if it is one that I'll have to use a REPL to update. And also not protected from anyone else who I share the canvas with.
>- Being able to temporarily disable trace showing. Sometimes I find I'm mousing around and the traces are quickly changing and showing different things making the visual field unstable. At times I just want to focus on the code and this can be a bit noisy and distracting.
>-How do I enter a new line when I'm at the end of the current line on a pipe statement? I have to go to the next line and to its beginning and then click enter.

**Comment by snasirca** [comment]
This feedback was from me. I'm looking forward to how this can be addressed.

**Comment by StachuDotNet** [comment]
Are you looking to have your endpoints secured so that only you have access to them, not concerned about 'users' yet? I'd like that too - so far, I've set up some really dumb auth for myself.

Since Dark login functionality is powered by Auth0, I wonder if Dark could ship a StdLib/package fn along the lines of `DarkAuth::isCanvasOwner` or `DarkAuth::hasAccessToCanvas` or `DarkAuth::failIfNotCanvasOwner` which you could use?

**Reply by snasirca** [reply]
I was looking to secure the endpoints for users. Like if someone logs in to my site, I want them to have access to all the secured endpoints like their profile, account, their dashboard, etc. And I want to have a way to add an "auth middleware" to each of those backend endpoints.
